diff -ur hamlib-4.6-orig/configure.ac hamlib-4.6/configure.ac
--- hamlib-4.6-orig/configure.ac	2024-12-24 00:55:54
+++ hamlib-4.6/configure.ac	2025-01-05 15:42:44
@@ -317,7 +317,7 @@
 
     [mingw* | pw32* | cygwin*], [
 	AM_CONDITIONAL([VERSIONDLL], true)
-	WINLDFLAGS="-Wl,--output-def,libhamlib.def -Wl,--add-stdcall-alias"
+	WINLDFLAGS="-Wl,--output-def,libhamlib.def"
     VERSIONDLL="version_dll.rc"
 	AC_SUBST([WINLDFLAGS])
 
diff -ur hamlib-4.6-orig/rigs/dummy/flrig.c hamlib-4.6/rigs/dummy/flrig.c
--- hamlib-4.6-orig/rigs/dummy/flrig.c	2024-12-24 00:55:54
+++ hamlib-4.6/rigs/dummy/flrig.c	2025-01-05 15:42:59
@@ -73,7 +73,7 @@
 static int flrig_set_vfo(RIG *rig, vfo_t vfo);
 static int flrig_set_ptt(RIG *rig, vfo_t vfo, ptt_t ptt);
 static int flrig_get_ptt(RIG *rig, vfo_t vfo, ptt_t *ptt);
-static int flrig_set_func(RIG *rig, vfo_t vfo, setting_t setting, int status);
+HAMLIB_EXPORT(int) flrig_set_func(RIG *rig, vfo_t vfo, setting_t setting, int status);
 static int flrig_set_split_freq(RIG *rig, vfo_t vfo, freq_t tx_freq);
 static int flrig_get_split_freq(RIG *rig, vfo_t vfo, freq_t *tx_freq);
 static int flrig_set_split_vfo(RIG *rig, vfo_t vfo, split_t split,
