diff --git a/src/msw/graphics.cpp b/src/msw/graphics.cpp
index abca42a35..c1079fedd 100644
--- a/src/msw/graphics.cpp
+++ b/src/msw/graphics.cpp
@@ -2597,7 +2597,7 @@ wxGraphicsRenderer* wxGraphicsRenderer::GetGDIPlusRenderer()
 
 wxGraphicsRenderer* wxGraphicsRenderer::GetDefaultRenderer()
 {
-    return wxGraphicsRenderer::GetGDIPlusRenderer();
+    return wxGraphicsRenderer::GetDirect2DRenderer();
 }
 
 bool wxGDIPlusRenderer::EnsureIsLoaded()
diff --git a/src/msw/graphicsd2d.cpp b/src/msw/graphicsd2d.cpp
index 89b74102a..e2a96a760 100644
--- a/src/msw/graphicsd2d.cpp
+++ b/src/msw/graphicsd2d.cpp
@@ -4767,7 +4767,8 @@ void wxD2DContext::DoDrawText(const wxString& str, wxDouble x, wxDouble y)
     GetRenderTarget()->DrawTextLayout(
         D2D1::Point2F(x, y),
         textLayout,
-        fontData->GetBrushData().GetBrush());
+        fontData->GetBrushData().GetBrush(),
+        D2D1_DRAW_TEXT_OPTIONS::D2D1_DRAW_TEXT_OPTIONS_ENABLE_COLOR_FONT);
 }
 
 void wxD2DContext::EnsureInitialized()
diff --git a/src/msw/listctrl.cpp b/src/msw/listctrl.cpp
index 538a7af59..6d5306799 100644
--- a/src/msw/listctrl.cpp
+++ b/src/msw/listctrl.cpp
@@ -37,6 +37,7 @@
 
 #include "wx/imaglist.h"
 #include "wx/vector.h"
+#include "wx/graphics.h"
 
 #include "wx/msw/private.h"
 #include "wx/msw/private/customdraw.h"
@@ -3095,7 +3096,13 @@ bool HandleSubItemPrepaint(LPNMLVCUSTOMDRAW pLVCD, HFONT hfont, int colCount)
     }
     //else: failed to get alignment, assume it's DT_LEFT (default)
 
-    DrawText(hdc, text, -1, &rc, fmt);
+    wxGraphicsRenderer* renderer = wxGraphicsRenderer::GetDirect2DRenderer();
+    wxGraphicsContext* context = renderer->CreateContextFromNativeContext(hdc);
+    context->DrawText(text, rc.left, rc.top);
+    delete context;
+    fprintf(stderr, "left: %ld, top: %ld\n", rc.left, rc.top);
+
+    //DrawText(hdc, text, -1, &rc, fmt);
 
     return true;
 }
